packages/block-editor/src/hooks/test/utils.js

```diff
@@ -133,15 +130,49 @@ describe( 'anchor', () => {
 			expect( settings.attributes ).toBe( undefined );
 		} );
 
-		it( 'should assign a new anchor attribute', () => {
-			const settings = registerBlockType( {
-				...blockSettings,
-				supports: {
-					anchor: true,
+		it( 'should assign a new anchor attribute depending on the value of the save property', () => {
+			const saveDefinitions = [
+				// The anchor attribute should be stored as a comment delimiter.
+				{
+					value: null,
+					schema: {
+						type: 'string',
+					},
 				},
-			} );
+				{
+					value: () => null,
+					schema: {
+						type: 'string',
+					},
+				},
+				{
+					value: undefined,
+					schema: {
+						type: 'string',
+					},
+				},
+				// The anchor attributes should be saved as part of the markup.
+				{
+					value: () => <div className="default" />,
+					schema: {
+						attribute: 'id',
+						selector: '*',
+						source: 'attribute',
+						type: 'string',
+					},
+				},
+			];
 
-			expect( settings.attributes ).toHaveProperty( 'anchor' );
+			saveDefinitions.forEach( ( { value, schema } ) => {
+				const settings = registerBlockType( {
+					...blockSettings,
+					save: value,
+					supports: {
+						anchor: true,
+					},
+				} );
+				expect( settings.attributes.anchor ).toEqual( schema );
+			} );
```

Comment:
In the unit test, it is checked whether the `anchor` attribute with the correct definition is registered according to the value of the `save` property.